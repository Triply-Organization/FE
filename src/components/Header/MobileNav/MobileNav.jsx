import { Button } from 'antd';
import { Typography } from 'antd';
import { PropTypes } from 'prop-types';
import React, { useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import { FaBars } from 'react-icons/fa';
import { VscChromeClose } from 'react-icons/vsc';
import { Link } from 'react-router-dom';

import './MobileNav.scss';

const { Title } = Typography;
export default function MobileNav(props) {
  const user = JSON.parse(localStorage.getItem('user')) || null;
  const token = localStorage.getItem('token');

  console.log(user);
  console.log(token);
  const { t } = useTranslation();
  const {
    mobileNavStatus,
    onChangeNavbarStatus,
    navItem,
    // eslint-disable-next-line no-unused-vars
    userDataLoginMobile,
  } = props;
  useEffect(() => {
    const mobileNav = document.querySelector('.mobileNav__overlays');
    if (mobileNavStatus) {
      mobileNav.classList.add('mobileNav__overlays--transition');
    } else {
      mobileNav.classList.remove('mobileNav__overlays--transition');
    }
  }, [mobileNavStatus]);
  return (
    <div className="mobileNav__wrapper">
      <div
        className="mobileNav__icon"
        onClick={() => {
          onChangeNavbarStatus();
        }}
      >
        <FaBars />
      </div>
      <div className="mobileNav__overlays">
        <div className="mobileNav_close-btn-wrapper">
          <button
            className="mobileNav_close-btn"
            onClick={() => {
              onChangeNavbarStatus();
            }}
          >
            <VscChromeClose />
          </button>
        </div>

        {token ? (
          <div className="mobileNav__account">
            <Title className="mobileNav__account-heading" level={4}>
              Hi {user?.name}
            </Title>
            <ul className="mobileNav__list">
              {userDataLoginMobile.map((item, index) => {
                return (
                  <li
                    onClick={item.onClick}
                    className="mobileNav__item"
                    key={index}
                  >
                    <Link to={item.link} className="mobileNav__link">
                      <span className="mobileNav__icon">{item.icon}</span>
                      <span className="mobileNav__words">{item.title}</span>
                    </Link>
                  </li>
                );
              })}
            </ul>
          </div>
        ) : (
          <div className="mobileNav__account">
            <Title className="mobileNav__account-heading" level={4}>
              {t('mobileNav.my_account')}
            </Title>
            <div className="mobileNav__account-btn">
              <Button type="primary" size="large">
                {t('mobileNav.login')}
              </Button>
              <Button type="default" size="large">
                {t('mobileNav.register')}
              </Button>
            </div>
          </div>
        )}

        <div className="mobileNav__list-wrapper">
          <ul className="mobileNav__list">
            {navItem.map((item, index) => {
              return (
                <li className="mobileNav__item" key={index}>
                  <Link to={'#'} className="mobileNav__link">
                    <span className="mobileNav__icon">{item.icon}</span>
                    <span className="mobileNav__words">{item.title}</span>
                  </Link>
                </li>
              );
            })}
          </ul>
        </div>
      </div>
    </div>
  );
}

MobileNav.propTypes = {
  mobileNavStatus: PropTypes.any,
  onChangeNavbarStatus: PropTypes.func,
  navItem: PropTypes.array,
  userDataLoginMobile: PropTypes.array,
};
