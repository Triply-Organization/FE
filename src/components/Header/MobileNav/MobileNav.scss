.mobileNav__wrapper {
}

.mobileNav__icon {
  font-size: 1.4rem;
  line-height: 1.4rem;
  padding: 5px 10px;
  cursor: pointer;
}

.mobileNav__overlays {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: #fff;
  transform: translateX(100%);
  opacity: 0;
  transition: all 0.25s ease;
}

.mobileNav__overlays--transition {
  transform: translateX(0%);
  opacity: 1;
}

.mobileNav_close-btn-wrapper {
  text-align: right;
}

.mobileNav_close-btn {
  padding: 10px;
  border: none;
  font-size: 1.5rem;
  line-height: 1.5rem;
  cursor: pointer;
  outline: none;
  background: #fff;
}

.mobileNav__account {
  padding: 0 18px 24px;

  & > .mobileNav__account-heading {
    text-align: left;
  }
}

.mobileNav__account-btn {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 12px;

  & > button {
    width: 100%;
  }
}

.mobileNav__list-wrapper {
  padding: 24px 18px 0;
  border-top: 1px solid rgba(204, 204, 204, 0.45);
}

.mobileNav__list {
  list-style-type: none;
}

.mobileNav__item {
  font-size: 1rem;
  text-transform: capitalize;
  text-align: left;
  padding: 6px 12px;
  border-radius: 5px;
  margin: 4px 0;
  cursor: pointer;
  display: flex;
  align-items: center;
  transition: all 0.2s ease-out;

  &:hover {
    background: #f0f0f0;
  }
}

.mobileNav__icon {
  font-size: 1.6rem;
  line-height: 1.6rem;
  padding: 5px 10px;
  cursor: pointer;
  transform: translateY(2px);
}

.mobileNav__words {
}

@media (max-width: 468px) {
  .mobileNav__account-btn {
    flex-direction: column;
  }
}
