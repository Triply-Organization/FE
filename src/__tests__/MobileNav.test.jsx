import {
  act,
  fireEvent,
  getByText,
  render,
  screen,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import { describe } from 'vitest';

import MobileNav from '../components/Header/MobileNav';

describe('MobileNav component', () => {
  it('should render correctly', () => {
    const result = render(
      <BrowserRouter>
        <MobileNav />
      </BrowserRouter>,
    );
    expect(result).toMatchSnapshot();
  });
});
