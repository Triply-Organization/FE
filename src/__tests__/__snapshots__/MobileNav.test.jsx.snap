// Vitest Snapshot v1

exports[`MobileNav component > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mobileNav__wrapper"
      >
        <div
          class="mobileNav__icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            />
          </svg>
        </div>
        <div
          class="mobileNav__overlays"
        >
          <div
            class="mobileNav_close-btn-wrapper"
          >
            <button
              class="mobileNav_close-btn"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M7.116 8l-4.558 4.558.884.884L8 8.884l4.558 4.558.884-.884L8.884 8l4.558-4.558-.884-.884L8 7.116 3.442 2.558l-.884.884L7.116 8z"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
          <div
            class="mobileNav__account"
          >
            <h4
              class="ant-typography mobileNav__account-heading"
            >
              mobileNav.my_account
            </h4>
            <div
              class="mobileNav__account-btn"
            >
              <button
                class="ant-btn ant-btn-primary ant-btn-lg"
                type="button"
              >
                <a
                  href="/login"
                >
                  mobileNav.login
                </a>
              </button>
              <button
                class="ant-btn ant-btn-default ant-btn-lg"
                type="button"
              >
                <a
                  href="/register"
                >
                  mobileNav.register
                </a>
              </button>
            </div>
          </div>
          <div
            class="mobileNav__list-wrapper"
          >
            <ul
              class="mobileNav__list"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mobileNav__wrapper"
    >
      <div
        class="mobileNav__icon"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          />
        </svg>
      </div>
      <div
        class="mobileNav__overlays"
      >
        <div
          class="mobileNav_close-btn-wrapper"
        >
          <button
            class="mobileNav_close-btn"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M7.116 8l-4.558 4.558.884.884L8 8.884l4.558 4.558.884-.884L8.884 8l4.558-4.558-.884-.884L8 7.116 3.442 2.558l-.884.884L7.116 8z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          class="mobileNav__account"
        >
          <h4
            class="ant-typography mobileNav__account-heading"
          >
            mobileNav.my_account
          </h4>
          <div
            class="mobileNav__account-btn"
          >
            <button
              class="ant-btn ant-btn-primary ant-btn-lg"
              type="button"
            >
              <a
                href="/login"
              >
                mobileNav.login
              </a>
            </button>
            <button
              class="ant-btn ant-btn-default ant-btn-lg"
              type="button"
            >
              <a
                href="/register"
              >
                mobileNav.register
              </a>
            </button>
          </div>
        </div>
        <div
          class="mobileNav__list-wrapper"
        >
          <ul
            class="mobileNav__list"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
